python_modules/libraries/dagster-databricks/dagster_databricks/ext.py

```diff
@@ -0,0 +1,165 @@
+    """Ext client for databricks.
+
+    Args:
+        client (WorkspaceClient): A databricks workspace client.
+        env (Optional[Mapping[str,str]]: An optional dict of environment variables to pass to the databricks job.
+    """
+
+    env: Optional[Mapping[str, str]] = Field(
+        default=None,
+        description="An optional dict of environment variables to pass to the subprocess.",
+    )
+
+    def __init__(
+        self,
+        client: WorkspaceClient,
+        env: Optional[Mapping[str, str]] = None,
+        context_injector: Optional[ExtContextInjector] = None,
+        message_reader: Optional[ExtMessageReader] = None,
+    ):
+        self.client = client
+        self.env = env
+        self.context_injector = check.opt_inst_param(
+            context_injector,
+            "context_injector",
+            ExtContextInjector,
+        ) or ExtDbfsContextInjector(client=self.client)
+        self.message_reader = check.opt_inst_param(
+            message_reader,
+            "message_reader",
+            ExtMessageReader,
+        ) or ExtDbfsMessageReader(client=self.client)
+
+    def run(
+        self,
+        task: jobs.SubmitTask,
+        *,
+        context: OpExecutionContext,
+        extras: Optional[ExtExtras] = None,
+        submit_args: Optional[Mapping[str, str]] = None,
+    ) -> None:
+        """Run a Databricks job with the EXT protocol.
+
+        Args:
+            task (databricks.sdk.service.jobs.SubmitTask): Specification of the databricks
+                task to run. Environment variables used by dagster-ext will be set under the
+                `spark_env_vars` key of the `new_cluster` field (if there is an existing dictionary
+                here, the EXT environment variables will be merged in). Everything else will be
+                passed unaltered under the `tasks` arg to `WorkspaceClient.jobs.submit`.
+            submit_args (Optional[Mapping[str, str]]): Additional keyword arguments that will be
+                forwarded as-is to `WorkspaceClient.jobs.submit`.
+        """
+        with ext_protocol(
+            context=context,
+            extras=extras,
+            context_injector=self.context_injector,
+            message_reader=self.message_reader,
+        ) as ext_context:
+            submit_task_dict = task.as_dict()
+            submit_task_dict["new_cluster"]["spark_env_vars"] = {
+                **submit_task_dict["new_cluster"].get("spark_env_vars", {}),
+                **(self.env or {}),
+                **ext_context.get_external_process_env_vars(),
+            }
+            task = jobs.SubmitTask.from_dict(submit_task_dict)
+            run_id = self.client.jobs.submit(
+                tasks=[task],
+                **(submit_args or {}),
+            ).bind()["run_id"]
+
+            while True:
+                run = self.client.jobs.get_run(run_id)
+                context.log.info(f"Run state: {run.state.life_cycle_state}")
```

Comment:
Let's make this a more information message. Someone who is just viewing the run but has not read the code should be able to understand what is going on.

f"Current run state of databricks run {run_id}: {run.state.life_cycle_state}"

Would be cool to render a url the points to databricks as well, but that is probably context-specific?

Reply:
Changed the message along suggested lines.

I'm not sure how to get the URL at present and trying to get this in for release this AM.